internal/pkg/manager/nodestatus/nodestatus.go

```diff
@@ -167,12 +169,28 @@ func (r *StatusReconciler) Reconcile(_ context.Context, req reconcile.Request) (
 	// make sure we have all the statuses already
 	hasStatuses := len(nodeStatusList.Items)
 	wantsStatuses := spodDS.Status.DesiredNumberScheduled
-	if wantsStatuses != int32(hasStatuses) {
+	if wantsStatuses > int32(hasStatuses) {
 		logger.Info("Not updating policy: not all statuses are ready",
 			"has", hasStatuses, "wants", wantsStatuses)
 		// Don't reconcile again, let's just wait for another update
 		return reconcile.Result{}, nil
 	}
+	if wantsStatuses < int32(hasStatuses) {
+		// this happens when nodes are removed from the cluster
+		logger.Info("Removing extra statuses", "has", hasStatuses, "wants", wantsStatuses)
+		nodeName, err := r.removeStatusForDeletedNode(ctx, nodeStatusList, lprof)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("cannot remove extra statuses: %w", err)
+		}
+		if nodeName != "" {
+			// remove the deleted node finalizer string from the profile
+			logger.Info("Removing finalizer from profile", "profile", prof.GetName(), "node", nodeName)
+			if err := util.RemoveFinalizer(ctx, r.client, prof, util.GetFinalizerNodeString(nodeName)); err != nil {
+				return reconcile.Result{}, fmt.Errorf("cannot remove finalizer from profile: %w", err)
+			}
+		}
+		return reconcile.Result{}, nil
```

Comment:
since this code would only work for one node removal, should we force another reconcile loop here by returning:
```
return reconcile.Result{Requeue: true}, nil
```
this would allow us to both keep the logic simple to just remove one finalizer and one node object for 99% case where just one node goes away, but it would also cover the case where more nodes go away through multiple reconcile attempts.